addons/l10n_es_edi_facturae/models/l10n_es_edi_facturae_certificate.py

```diff
@@ -76,10 +76,13 @@ def _sign_xml(self, edi_data, signature_data):
         rfc4514_attr = dict(element.rfc4514_string().split("=", 1) for element in cert_public.issuer.rdns)
 
         # The 'Organizational Unit' field is optional
+        issuer = f"CN={rfc4514_attr.pop('CN')}, "
         if 'OU' in rfc4514_attr:
-            issuer = f"CN={rfc4514_attr['CN']}, OU={rfc4514_attr['OU']}, O={rfc4514_attr['O']}, C={rfc4514_attr['C']}"
-        else:
-            issuer = f"CN={rfc4514_attr['CN']}, O={rfc4514_attr['O']}, C={rfc4514_attr['C']}"
+            issuer += f"OU={rfc4514_attr.pop('OU')}, "
+        issuer += f"O={rfc4514_attr.pop('O')}, C={rfc4514_attr.pop('C')}"
+
+        # Add remaining certificate fields (not all certificates have other fields)
+        issuer += "".join([f", {key}={value}" for key, value in rfc4514_attr.items()])
```

Comment:
If "the order of these fields does not affect the validation process" and that we should include all certificates, shouldn't this be enough ?

```suggestion
        issuer = ",".join([f"{key}={value}" for key, value in rfc4514_attr.items()])
```

Reply:
Hello, when I say “the order of these fields,” I am only referring to the new fields that I am adding (fields left). The other fields have a well-defined order, see the PR: https://github.com/odoo/odoo/pull/169977.

Reply:
Okay makes sense then !